src/InstrumentationEngine/ProfilerManagerForInstrumentationMethod.cpp

```diff
@@ -29,7 +29,13 @@ CProfilerManagerForInstrumentationMethod::CProfilerManagerForInstrumentationMeth
         // Get InstrumentationMethod-specific LogLevel Environment Variable
         tstring wszInstrumentationMethodLogLevelEnvVar = _T("MicrosoftInstrumentationEngine_LogLevel_") + m_wszInstrumentationMethodGuid;
         WCHAR wszEnvVar[MAX_PATH];
```

Comment:
Could you refactor this so that we only look up the environment variable if the engine doesn't know the specific value? e.g.

if (get from profiler manager) {
}
else
{
  lookup from env var
  if (env exists)
  {
    get from env var
  }
  else
  {
    get from global
  }
}